src/main/flight/pid.c

```diff
@@ -52,6 +52,10 @@
 
 extern uint8_t motorCount;
 extern float dT;
+extern bool motorLimitReached;
+bool antiWindupProtection;
+
+#define PREVENT_WINDUP(x,y,z) { if (z) {if (ABS(x) > ABS(y)) { if (x < 0) { x = -ABS(y); }  else { x = ABS(y); } } } else { y = x; }}
```

Comment:
Lol.....didn't realise it was that hard to read :D

I hated to put long if else statements in all pid controllers so put it in the macro. I should indeed better commented it more.

Good point about storing the absolute values.